press/utils/__init__.py

```diff
@@ -543,3 +545,15 @@ def poly_get_doctype(doctypes, name):
 		if frappe.db.exists(doctype, name):
 			return doctype
 	return doctypes[-1]
+
+
+def reconnect_on_failure():
+	@wrapt.decorator
+	def wrapper(wrapped, instance, args, kwargs):
+		try:
+			return wrapped(*args, **kwargs)
+		except InterfaceError:
+			frappe.db.connect()
```

Comment:
Also you should add a retry counter here, this can cause infinite spam to DB :sweat_smile:

Reply:
I'm not sure what you mean, this quits after a single retry. If the retry throws, it isn't handled. Am I missing something? 🤔

Do you mean after multiple invocations of the wrapper across different functions? Even then, given that only one of them would run at a time, only two tries would occur.

Reply:
Got it, I thought it was recursively calling same function with `except` block